### PR TITLE
sdl2_image 2.0.5

### PR DESCRIPTION
**Pull request by ericLemanissier**
this is a port of https://github.com/bincrafters/community/tree/main/recipes/sdl2_image

Specify library name and version:  **sdl2_image/2.0.5**

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Croydon** [comment]
Let's rename the folder to `2.0.5` as the next version has official CMake support

File: recipes/sdl_image/all/conanfile.py [hunk 1]
```diff
@@ -100,29 +100,30 @@ def _configure_cmake(self):
...
+        self._cmake.definitions["JPG_DYNAMIC"] = self.options["libjpeg"].shared if self.options.with_libjpeg else False
+        self._cmake.definitions["PNG_DYNAMIC"] = self.options["libpng"].shared if self.options.with_libpng else False
+        self._cmake.definitions["WEBP_DYNAMIC"] = self.options["libwebp"].shared if self.options.with_libwebp else False
         self._cmake.definitions["SDL_IS_SHARED"] = self.options["sdl"].shared
 
         self._cmake.configure(build_dir="build")
         return self._cmake
 
     def build(self):
+        tools.rmdir(os.path.join(self._source_subfolder, "external"))
```
Comment (madebr) [inline recipes/sdl_image/all/conanfile.py:126]: sdl_image uses a vendored header-only `minz.h`.
I tried replacing it with our `miniz/2.2.0`.
But it looks like the API changed between 1.x and 2.x.

**Comment by madebr** [comment]
> Let's rename the folder to `2.0.5` as the next version has official CMake support

I just tried their current master with the cmake script of this pr, and it *just works*.
I propose to add a patch to 2.0.5 that creates a compatible cmake script.

**Comment by madebr** [comment]
`sdl_image:shared` must be the same as `sdl::shared`.
Or, we patch sdl_image.

**Comment by madebr** [comment]
Hey guys,
while working on this pr I opened a pr at the upstream SDL_image repo: https://github.com/libsdl-org/SDL_image/pull/189

It would be nice if (some of ) you gave some constructive feedback on what/how to improve.

**Comment by SSE4** [comment]
does anything block it from making ready for the review?

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by madebr** [comment]
> > Let's rename the folder to `2.0.5` as the next version has official CMake support
> 
> This comment seems worthy

When this new sdl_image release happens, I would add our cmake script as a patch to make it behave the same as the upstream cmake project (to 2.0.5 only).
See https://github.com/libsdl-org/SDL_image/pull/189
It's very similar to ours.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**